app/models/site.rb

```diff
@@ -28,16 +28,33 @@ def user_fields
     UserField.order(:position).all
   end
 
-  def categories
-    @categories ||= begin
-      categories = Category
+  CATEGORIES_CACHE_KEY = "site_categories"
+
+  def self.clear_cache
+    Discourse.cache.delete(CATEGORIES_CACHE_KEY)
+  end
+
+  def self.all_categories_cache
+    # Categories do not change often so there is no need for us to run the
+    # same query and spend time creating ActiveRecord objects for every requests.
+    #
+    # Do note that any new association added to the eager loading needs a
+    # corresponding ActiveRecord callback to clear the categories cache.
```

Comment:
Shouldn't we also clear the cache when we change the logo/background of a category?

Reply:
The foreign keys for uploads are on the categories table itself so the callback on the category model will be triggered when the uploads change.